hecuba/hdict.py

```diff
@@ -437,18 +431,23 @@ def make_persistent(self, name):
             log.warn("Error creating the StorageDict table: %s %s", query_table, ex)
             raise ex
         key_names = map(lambda a: a[0].encode('UTF8'), self._primary_keys)
-        column_names = self._columns
+        values_names = self._columns
 
         self._hcache_params = (self._ksp, self._table,
                                self._storage_id,
-                               self._tokens, key_names, map(lambda x: {"name": x[0], "type": x[1]}, column_names),
+                               self._tokens, key_names, map(lambda x: {"name": x[0], "type": x[1]}, values_names),
                                {'cache_size': config.max_cache_size,
                                 'writer_par': config.write_callbacks_number,
                                 'write_buffer': config.write_buffer_size})
         log.debug("HCACHE params %s", self._hcache_params)
         self._hcache = Hcache(*self._hcache_params)
         # Storing all in-memory values to cassandra
         for key, value in dict.iteritems(self):
+            if issubclass(value.__class__, IStorage):
+                # new name as ksp.table_valuename, where valuename is either defined by the user or set by hecuba
+                val_name = self._ksp + '.' + self._table + '_' + self._columns[0][0]
+                value.make_persistent(val_name)
+                value = value._storage_id
             self._hcache.put_row(self._make_key(key), self._make_value(value))
         if hasattr(self, '_indexed_args') and self._indexed_args is not None:
             index_query = 'CREATE CUSTOM INDEX IF NOT EXISTS ' + self._table + '_idx ON '
```

Comment:
add a self.__dict__ = {} to clean the memory dict after make the StorageDict persistent

Reply:
add tests to check it work (try with a storageObj with a custom function)